pkg/utils/tracing/opentelemetry.go

```diff
@@ -0,0 +1,56 @@
+package tracing
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type OpenTelemetryTracer struct {
+	realTracer trace.Tracer
+}
+
+func NewOpenTelemetryTracer(t trace.Tracer) Tracer {
+	return &OpenTelemetryTracer{
+		realTracer: t,
+	}
+}
+
+func (t OpenTelemetryTracer) StartSpan(operationName string) Span {
+	_, realspan := t.realTracer.Start(context.Background(), operationName)
+	return openTelemetrySpan{realSpan: realspan}
+}
+
+func (t OpenTelemetryTracer) StartSpanFromTraceParent(operationName string, parentTraceId, parentSpanId string) Span {
```

Comment:
```suggestion
func (t OpenTelemetryTracer) StartSpanFromTraceParent(ctx context.Context, operationName string, parentTraceId, parentSpanId string) Span {
```

Let's propagate context. This is good practice.